### PR TITLE
Organize internationalized documentation

### PR DESCRIPTION
**Pull request by patricklodder**
Implements proposal from #3436

1. Creates a new structure in `doc/intl/`, where each language gets their own folder and a central `README.md` that lists all translated documents
2. Moves current translated README files into this structure
3. Removes cross linking, introduces hierarchical linking

### CODE REVIEW

**Comment by slightlyskepticalpotat** [comment]
Missed this a few months ago—it definitely seems like an improvement from how we do things right now.

**Review by slightlyskepticalpotat — APPROVED** [review]

File: README.md [hunk 1]
```diff
@@ -11,7 +11,7 @@ Dogecoin Core [DOGE, Ð]
 
 </div>
 
-Select language: EN | [CN](./README_zh_CN.md) | [PT](./README_pt_BR.md) | [FA](./README_fa_IR.md) | [VI](./README_vi_VN.md)
+For internationalized documentation, see the index at [doc/intl](doc/intl/README.md)
```
[the remark this answers is not in the document]
Comment (patricklodder) [reply README.md]: Fixed.

**Comment by qlpqlp** [comment]
> Nice, looks good. Although, I'm thinking now too that there should also be a policy written somewhere explaining this part of your proposal:
> 
> > "... All translated documents are required to be up-to-date by release time, or they will be deleted and unlinked from the release (depending on which branching strategy we'll pursue). This prevents distributing outdated information with a release."
> 
> Perhaps it would be good to write that in `doc/intl/README.md`, as I think that's probably the most likely place for contributors and maintainers to look.

If you also want to help organise the documentation here https://dogecoincore.com/ 
I will appreciate it, you can do so easily here https://github.com/qlpqlp/dogecoin-core-docs/tree/main/
Just clone and on **src/** folder you will see all languages, but for now only **src/en/** is all documented

I don't mind to move this repo to **dogecoin** repo because it will only contain CORE documentation :)

**Comment by qlpqlp** [comment]
> @qlpqlp I don't have time for any more commitments atm, sorry fren.

No worries <3, I will then update wen I have some more motivation :) (time I always find out, motivation is the only barrier for me at the moment lol)

**Review by chromatic — COMMENTED** [review]
utACK. Squash this and I think it's good to go.

**Comment by patricklodder** [comment]
Rebased, redid the move of the now newer fa_IR readme, and squashed the punctuation fixup.

**Review by chromatic — APPROVED** [review]
ACK, commits are good and all tests pass on x86-64 Linux.

**Pull request merged**